### ISSUE TITLE
WARNING - Warn: Using legacy field 'p.controlnet_input_image'

### ISSUE THREAD

**Issue by k52252467** [ClockZinc/sd-webui-IS-NET-pro#16]
ISnet::MFR::will process  501 images
ISNET::MFR::Single mode OPEN!!!
2024-03-29 15:40:28,281 - ControlNet - INFO - unit_separate = False, style_align = False
2024-03-29 15:40:28,469 - ControlNet - INFO - Loading model: control_v11f1e_sd15_tile [a371b31b]
2024-03-29 15:40:28,855 - ControlNet - INFO - Loaded state_dict from [C:\stable-diffusion-webui\models\ControlNet\control_v11f1e_sd15_tile.pth]
2024-03-29 15:40:28,855 - ControlNet - INFO - controlnet_default_config
2024-03-29 15:40:29,990 - ControlNet - INFO - ControlNet model control_v11f1e_sd15_tile [a371b31b](ControlModelType.ControlNet) loaded.
2024-03-29 15:40:29,991 - ControlNet - WARNING - Warn: Using legacy field 'p.controlnet_input_image'
2024-03-29 15:40:30,027 - ControlNet - INFO - Using preprocessor: tile_colorfix+sharp
2024-03-29 15:40:30,027 - ControlNet - INFO - preprocessor resolution = 792
2024-03-29 15:40:30,032 - ControlNet - INFO - Loading model: control_v11p_sd15_lineart [43d4be0d]
2024-03-29 15:40:30,256 - ControlNet - INFO - Loaded state_dict from [C:\stable-diffusion-webui\models\ControlNet\control_v11p_sd15_lineart.pth]
2024-03-29 15:40:30,256 - ControlNet - INFO - controlnet_default_config
2024-03-29 15:40:31,265 - ControlNet - INFO - ControlNet model control_v11p_sd15_lineart [43d4be0d](ControlModelType.ControlNet) loaded.
2024-03-29 15:40:31,265 - ControlNet - WARNING - Warn: Using legacy field 'p.controlnet_input_image'
2024-03-29 15:40:31,300 - ControlNet - INFO - Using preprocessor: lineart_standard
2024-03-29 15:40:31,300 - ControlNet - INFO - preprocessor resolution = 792
2024-03-29 15:40:31,348 - ControlNet - INFO - Loading model: control_v11f1p_sd15_depth [cfd03158]
2024-03-29 15:40:31,560 - ControlNet - INFO - Loaded state_dict from [C:\stable-diffusion-webui\models\ControlNet\control_v11f1p_sd15_depth.pth]
2024-03-29 15:40:31,560 - ControlNet - INFO - controlnet_default_config
2024-03-29 15:40:32,570 - ControlNet - INFO - ControlNet model control_v11f1p_sd15_depth [cfd03158](ControlModelType.ControlNet) loaded.
2024-03-29 15:40:32,570 - ControlNet - WARNING - Warn: Using legacy field 'p.controlnet_input_image'
2024-03-29 15:40:32,606 - ControlNet - INFO - Using preprocessor: depth_anything
2024-03-29 15:40:32,606 - ControlNet - INFO - preprocessor resolution = 792
Using cache found in C:\Users\user/.cache\torch\hub\facebookresearch_dinov2_main
2024-03-29 15:40:36,048 - ControlNet - INFO - Loading model from cache: control_v11f1p_sd15_depth [cfd03158]
2024-03-29 15:40:36,048 - ControlNet - WARNING - Warn: Using legacy field 'p.controlnet_input_image'
2024-03-29 15:40:36,084 - ControlNet - INFO - Using preprocessor: depth_hand_refiner
2024-03-29 15:40:36,084 - ControlNet - INFO - preprocessor resolution = 792
set os.environ[OMP_NUM_THREADS] to 4
=> loading pretrained model C:\stable-diffusion-webui\extensions\sd-webui-controlnet\annotator\downloads\hand_refiner\hr16/ControlNet-HandRefiner-pruned\hrnetv2_w64_imagenet_pretrained.pth
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
2024-03-29 15:40:43,425 - ControlNet - INFO - Loading model: control_v11p_sd15_openpose [cab727d4]
2024-03-29 15:40:43,898 - ControlNet - INFO - Loaded state_dict from [C:\stable-diffusion-webui\models\ControlNet\control_v11p_sd15_openpose.pth]
2024-03-29 15:40:43,899 - ControlNet - INFO - controlnet_default_config
2024-03-29 15:40:44,982 - ControlNet - INFO - ControlNet model control_v11p_sd15_openpose [cab727d4](ControlModelType.ControlNet) loaded.
2024-03-29 15:40:44,982 - ControlNet - WARNING - Warn: Using legacy field 'p.controlnet_input_image'
2024-03-29 15:40:45,017 - ControlNet - INFO - Using preprocessor: openpose_faceonly
2024-03-29 15:40:45,017 - ControlNet - INFO - preprocessor resolution = 792
2024-03-29 15:40:47,609 - ControlNet - INFO - ControlNet Hooked - Time = 19.32997465133667

The warning message indicates the use of a deprecated field 'p.controlnet_input_image'. This means that certain parts of the code are utilizing functionality or methods that are no longer recommended.

[scripts/controlnet.py](https://github.com/Mikubill/sd-webui-controlnet/blob/2a210f0489a4484f55088159bbfa51aaf73e10d9/scripts/controlnet.py#L660)

657 | logger.warning("Warn: Using legacy field 'p.image_control'.")
658 | input_image = HWC3(np.asarray(p_image_control))
659 | elif p_input_image is not None:
660 | logger.warning("Warn: Using legacy field 'p.controlnet_input_image'")
661 | if isinstance(p_input_image, dict) and "mask" in p_input_image and "image" in p_input_image:
662 | color = HWC3(np.asarray(p_input_image['image']))
663 | alpha = np.asarray(p_input_image['mask'])

**Comment by ClockZinc** [comment]
The warning message indicates the use of a deprecated field 'p.controlnet_input_image'. This means that certain parts of the code are utilizing functionality or methods that are no longer recommended.
Maybe my isnet Version is too late for current SD controlnet
You can try to Fallback SD controlnet to previous version

**Comment by k52252467** [comment]
> The warning message indicates the use of a deprecated field 'p.controlnet_input_image'. This means that certain parts of the code are utilizing functionality or methods that are no longer recommended. Maybe my isnet Version is too late for current SD controlnet You can try to Fallback SD controlnet to previous version


Since I reinstalled my computer, I don't have versions from too long ago, only versions within the last 1 month.


Controlnet seems to only have version 1.1.436 available for separate download

https://github.com/Mikubill/sd-webui-controlnet/tags

This is a long way from when you last updated, and updating to such an early version that I might not be able to use some of the new Controlnet models

**Comment by ClockZinc** [comment]
maybe this video could help if you understand CHinese. The comments has some link.
https://www.bilibili.com/video/BV1iM4y1y7oA/?spm_id_from=333.999.0.0&vd_source=5e70e5e1970190a97f21d67edfe4f0c0

**Comment by ClockZinc** [comment]
then maybe you need to find an older-enough Version of SD and controlnet, if you really want to use this.
